components/02-elements/badge/Badge.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <component
+    :is="tag"
+    :class="['badge', {
+      'badge--new': newBadge,
+      'badge--discount': discountBadge
+    }]"
+  >
+    <slot/>
+  </component>
+</template>
+
+<script>
+export default {
+  props: {
+    tag: {
+      type: String,
+      default: 'div'
+    },
+    newBadge: {
```

Comment:
It's kinda weird to have have name of component in prop name, IMO it's perfectly fine to go just with `new` / `discount` or... define a `type` that will be `String` and add a simple validator that checks if value equals something real